lib/cocoapods/generator/copy_xcframework_script.rb

```diff
@@ -93,29 +99,15 @@ def script
   fi
   for i in ${!paths[@]}; do
     local matched_all_archs="1"
-    for target_arch in $target_archs
-    do
-      if ! [[ "${paths[$i]}" == *"$target_variant"* ]]; then
+    local slice_archs="$(archs_for_slice "${xcframework_name}/${paths[$i]}")"
+    local slice_variant="$(variant_for_slice "${xcframework_name}/${paths[$i]}")"
+    for target_arch in $target_archs; do
+      if ! [[ "${slice_variant}" == "$target_variant" ]]; then
         matched_all_archs="0"
         break
       fi
 
-      # Verifies that the path contains the variant string (simulator or maccatalyst) if the variant is set.
-      if [[ -z "$target_variant" && ("${paths[$i]}" == *"simulator"* || "${paths[$i]}" == *"maccatalyst"*) ]]; then
-        matched_all_archs="0"
-        break
-      fi
-
-      # This regex matches all possible variants of the arch in the folder name:
-      # Let's say the folder name is: ios-armv7_armv7s_arm64_arm64e/CoconutLib.framework
-      # We match the following: -armv7_, _armv7s_, _arm64_ and _arm64e/.
-      # If we have a specific variant: ios-i386_x86_64-simulator/CoconutLib.framework
-      # We match the following: -i386_ and _x86_64-
-      # When the .xcframework wraps a static library, the folder name does not include
-      # any .framework. In that case, the folder name can be: ios-arm64_armv7
-      # We also match _armv7$ to handle that case.
-      local target_arch_regex="[_\\-]${target_arch}([\\/_\\-]|$)"
-      if ! [[ "${paths[$i]}" =~ $target_arch_regex ]]; then
+      if ! echo "${slice_archs}" | tr " " "\\n" | grep -F -q -x "$target_arch"; then
```

Comment:
do you have some context on this change? it's unclear to me what this is doing. is it splitting the archs into separate lines and then using `grep` to see if any lines match?

Reply:
@amorde exactly - it's an internet-suggested way to check that `$target_arch` is in the `$slice_archs` list (which is space-separated).